src/main/flight/autopilot.c

```diff
@@ -91,14 +196,172 @@ void altitudeControl(float targetAltitudeCm, float taskIntervalS, float vertical
     DEBUG_SET(DEBUG_AUTOPILOT_ALTITUDE, 7, lrintf(altitudeF));
 }
 
-bool isBelowLandingAltitude(void)
+void setSticksActiveStatus(bool areSticksActive)
 {
-    return getAltitudeCm() < 100.0f * autopilotConfig()->landing_altitude_m;
+    posHold.sticksActive = areSticksActive;
+}
+
+void setTargetLocation(gpsLocation_t newTargetLocation) {
+    currentTargetLocation = newTargetLocation;
+}
+
+bool positionControl(void) {
+
+    // exit if we don't have a GPS 3D fix
+    if (!STATE(GPS_FIX)) {
+        return false; // cannot proceed without a GPS location
+    }
+    // and if no valid heading from GPS, or no mag
+    if (!canUseGPSHeading
+#ifdef USE_MAG
+        && !compassIsHealthy()
+#endif
+        ) {
+        return false;
+    }
+    
+    if (!wasThrottleRaised()) {
+        return false;
+    }
+
+    if (isNewDataForPosHold()) {
+        posHold.gpsDataIntervalS = getGpsDataIntervalSeconds(); // interval for current GPS data value 0.01s to 1.0s
+        posHold.gpsDataFreqHz = 1.0f / posHold.gpsDataIntervalS;
+        if (posHold.sticksActive) {
+            // if a Position Hold deadband is set, and sticks are outside deadband, allow pilot control in angle mode
+            resetPositionControlParams(&posHold.NS);
+            resetPositionControlParams(&posHold.EW);
+            posHold.pidSumRoll = 0.0f;
+            posHold.pidSumPitch = 0.0f;
+        } else {
+            // first get xy distances from current location (gpsSol.llh) to target location
+            float nsDistance; // cm, steps of 11.1cm, North of target is positive
+            float ewDistance; // cm, steps of 11.1cm, East of target is positive
+            GPS_distances(&gpsSol.llh, &currentTargetLocation, &nsDistance, &ewDistance);
+            float distanceCm = sqrtf(sq(nsDistance) + sq(ewDistance));
+    
+            posHold.NS.distance = nsDistance;
+            posHold.EW.distance = ewDistance;
+    
+            posHold.pt1Gain = pt1FilterGain(posHold.lpfCutoff, posHold.gpsDataIntervalS);
+            const float leak = 1.0f - 0.4f * posHold.gpsDataIntervalS; // gpsDataIntervalS is not more than 1.0s
```

Comment:
You can model it as RC/complementary filter with zero on input and use functions to convert time constant into RC alpha

Reply:
Yes this needs improvement.  
At this point I'm just monitoring the data to see if it is needed at all.